### ISSUE TITLE
Login next param is not respected but redirects to admin home

### ISSUE THREAD

**Issue by allcaps** [wagtail/wagtail#4698]
When I'm logged out and request an admin page the login url becomes:

    http://localhost:8000/admin/login/?next=/admin/pages/123/edit/

After login form submit, I expect to end up at `/admin/pages/123/edit/`.

But this:

https://github.com/wagtail/wagtail/blob/0161f2664b8968b6d5649074d3d044a312f1f1c9/wagtail/admin/views/account.py#L205-L206

Makes you always go to `/admin/`

Those two Wagtail `get_success_url` lines have to be dropped. Since Django's `LoginView.get_succes_url` is just fine.

/django/contrib/auth/views.py:92
```python
def get_redirect_url(self):
    """Return the user-originating redirect URL if it's safe."""
    redirect_to = self.request.POST.get(
        self.redirect_field_name,
        self.request.GET.get(self.redirect_field_name, '')
    )
    url_is_safe = is_safe_url(
        url=redirect_to,
        allowed_hosts=self.get_success_url_allowed_hosts(),
        require_https=self.request.is_secure(),
    )
    return redirect_to if url_is_safe else ''

def get_success_url(self):
    url = self.get_redirect_url()
    return url or resolve_url(settings.LOGIN_REDIRECT_URL)
```

This bug is introduced 4 months ago when we switched to class based auth views. Or maybe there is a reason to always go to `/admin/`?

**Comment by loicteixeira** [comment]
I don't see any reason not to respect the `next` parameter and looking at the commits, there are no indications of the change being intentional. Thanks for the report @allcaps.

**Comment by loicteixeira** [comment]
Thanks for the report @allcaps. Fix merged in aac33358834ce77e221b0b26db1af36898a7e9c8.
